src/main/java/knightminer/inspirations/common/block/HidableBlock.java

```diff
@@ -0,0 +1,30 @@
+package knightminer.inspirations.common.block;
+
+import knightminer.inspirations.common.IHidable;
+import net.minecraft.block.Block;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+
+import javax.annotation.Nonnull;
+import java.util.function.Supplier;
+
+public class HidableBlock extends Block implements IHidable {
+    private final Supplier<Boolean> enabled;
+
+    public HidableBlock(Properties properties, Supplier<Boolean> isEnabled) {
+        super(properties);
+        this.enabled = isEnabled;
+    }
+
+    public boolean isEnabled() {
```

Comment:
Missing `@Override` for interface methods, ideally I always want either `@Override` or javadocs for functions